### PR TITLE
[SYCL][CUDA] Add sub-group info queries

### LINKED ISSUE #2922: [CUDA] Sub-group info is not supported
SYCL 2020 sub-group info queries are not supported by CUDA backend. See
https://github.com/intel/llvm/blob/65c719ddfc23e4e00ef4e2caebd0aa0e258112fd/sycl/plugins/cuda/pi_cuda.cpp#L3005-L3013

### PR DESCRIPTION
**Pull request by Pennycook**
Queries return 0 for compile-time information that is currently ignored when
generating PTX, and assume that warps only guarantee forward progress on Volta.

Signed-off-by: John Pennycook <john.pennycook@intel.com>

Closes #2922.

### CODE REVIEW

File: sycl/plugins/cuda/pi_cuda.cpp
Comment (bader) [inline sycl/plugins/cuda/pi_cuda.cpp:3052]:
```suggestion
    pi_kernel kernel, pi_device device, pi_kernel_sub_group_info param_name,
```
otherwise `\TODO Untie from OpenCL` is still relevant.

Comment (Pennycook) [reply sycl/plugins/cuda/pi_cuda.cpp]: Ah, I missed that.  Thanks, Alexey.

**Comment by Pennycook** [comment]
@bader, @intel/llvm-reviewers-cuda, @alexbatashev, @smaslov-intel - Is there anything more to do on this?

**Comment by bader** [comment]
> @bader, @intel/llvm-reviewers-cuda, @alexbatashev, @smaslov-intel - Is there anything more to do on this?

This PR is waiting on code owner review from @@intel/llvm-reviewers-cuda.
@romanovvlad, @smaslov-intel, please, take a look.

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Pull request merged**
